src/main/usecases/Encounter.java

```diff
@@ -298,24 +204,24 @@ public boolean isCompleted() {
     public String getHelp(Player player) {
         System.out.println("All commands will be of the form -- command: [param_type]=[param_name]");
         if (doingGeneric) {
-            if (this.genericPool.get(currGenericIndex) instanceof Spinnable) {
```

Comment:
as long as all these big changes work, we are good to go. did you test it?

Reply:
yes of course :)

Reply:
okay merging.